### PR TITLE
docs: update api reference documentation

### PR DESCRIPTION
**Pull request by dvarasani-crest**
This PR updates API reference documentation as below
-  add missing function implementation and its params.
-  correcting parameters doc for some functions
-  add documentation for cim-compliance-report

Ref: https://splunk.atlassian.net/browse/ADDON-75003

### CODE REVIEW

File: pytest_splunk_addon/cim_tests/test_templates.py [hunk 1]
```diff
@@ -56,6 +56,13 @@ def test_cim_required_fields(
             - Check that each required field is extracted in all of the events mapped with the data model.
             - Check that if there are inter dependent fields, either all fields should be
               extracted or none of them should be extracted.
+
+        Args:
+            splunk_search_util (SearchUtil): Object that helps to search on Splunk.
+            splunk_ingest_data (fixture): Unused but required to ensure data was ingested before running test
```
Comment (harshilgajera-crest) [inline pytest_splunk_addon/cim_tests/test_templates.py:62]: Can we remove this unused line as we are using it.

File: pytest_splunk_addon/cim_tests/test_templates.py [hunk 2]
```diff
@@ -56,6 +56,13 @@ def test_cim_required_fields(
...
+            splunk_setup (fixture): Unused but required to ensure that test environment was set up before running test
+            splunk_searchtime_cim_fields (fixture): Pytest parameter to test required cim field extraction
+            record_property (fixture): Document facts of test cases.
```
Comment (harshilgajera-crest) [inline pytest_splunk_addon/cim_tests/test_templates.py:65]: we can add that we document it provide more info in the test failure reports

File: pytest_splunk_addon/cim_tests/test_templates.py [hunk 3]
```diff
@@ -163,6 +170,13 @@ def test_cim_fields_not_allowed_in_search(
         """
         This test case checks the event_count for the cim fields of type ["not_allowed_in_search_and_props", "not_allowed_in_search"].
         - Expected event_count for these fields is zero.
+
+        Args:
+            splunk_ingest_data (fixture): Unused but required to ensure data was ingested before running test
+            splunk_search_util (SearchUtil): Object that helps to search on Splunk.
+            splunk_setup (fixture): Unused but required to ensure that test environment was set up before running test
```
Comment (harshilgajera-crest) [inline pytest_splunk_addon/cim_tests/test_templates.py:177]: same as above

File: pytest_splunk_addon/cim_tests/test_templates.py [hunk 4]
```diff
@@ -491,6 +513,13 @@ def test_eventtype_mapped_multiple_cim_datamodel(
     def test_cim_fields_recommended(
         self, splunk_dm_recommended_fields, splunk_searchtime_cim_fields_recommended
     ):
+        """
+        This test case check that all the recommended cim fields of datamodel mapped with event are exracted.
```
Comment (harshilgajera-crest) [inline pytest_splunk_addon/cim_tests/test_templates.py:517]: typo

File: pytest_splunk_addon/fields_tests/sample_parser.py [hunk 5]
```diff
@@ -37,6 +46,15 @@ def parse_sample_files(folder_path):
 
 
 def parse_file(filename):
+    """
+    Parse the xml sample file to get the lisst of events
```
Comment (harshilgajera-crest) [inline pytest_splunk_addon/fields_tests/sample_parser.py:50]: typo

File: pytest_splunk_addon/fields_tests/sample_parser.py [hunk 6]
```diff
@@ -269,11 +363,23 @@ def __init__(self, event_tag):
         self.exceptions = self.extract_key_value_xml("exceptions")
 
     def get_transport_type(self):
+        """
+        Function to get the transport type of the evnt
```
Comment (harshilgajera-crest) [inline pytest_splunk_addon/fields_tests/sample_parser.py:367]: Please check for small typos for this PR

File: pytest_splunk_addon/cim_tests/test_templates.py (hunk 1 above)
Comment (dvarasani-crest) [reply pytest_splunk_addon/cim_tests/test_templates.py]: Done

File: pytest_splunk_addon/cim_tests/test_templates.py (hunk 2 above)
Comment (dvarasani-crest) [reply pytest_splunk_addon/cim_tests/test_templates.py]: Done

File: pytest_splunk_addon/cim_tests/test_templates.py (hunk 3 above)
Comment (dvarasani-crest) [reply pytest_splunk_addon/cim_tests/test_templates.py]: Updated

File: pytest_splunk_addon/cim_tests/test_templates.py (hunk 4 above)
Comment (dvarasani-crest) [reply pytest_splunk_addon/cim_tests/test_templates.py]: fixed

File: pytest_splunk_addon/fields_tests/sample_parser.py (hunk 5 above)
Comment (dvarasani-crest) [reply pytest_splunk_addon/fields_tests/sample_parser.py]: fixed

File: pytest_splunk_addon/fields_tests/sample_parser.py (hunk 6 above)
Comment (dvarasani-crest) [reply pytest_splunk_addon/fields_tests/sample_parser.py]: Updated as per the comments and fixed typos.

**Review by harshilgajera-crest — APPROVED** [review]

**Pull request merged**